src/commands/archive-card.js

```diff
@@ -0,0 +1,178 @@
+"use strict";
+
+var List = require('term-list');
+  var prompt = require('prompt');
+
+var __ = function(program, output, logger, config, trello, translator) {
+
+    var trelloApiCommand = {};
+
+    trelloApiCommand.makeTrelloApiCall = function(options, onComplete) {
+        if (!options.match) {
+            options.match = 'exact';
+            logger.warning('--match parameter not specified, defaulting to exact title match');
+        }
+        logger.info("Adding card");
+
+        // Grab our boards etc
+        var boardId = translator.getBoardIdByName(options.board);
+        var listId = translator.getListIdByBoardNameAndListName(options.board, options.list);
+
+        // Get a list of cards on a board
+        trello.get("/1/boards/" + boardId + "/cards/visible", function(err, data) {
+
+            if (err) {
+                throw err;
+            }
+            if (data.indexOf("Trello is down") !== -1) {
+                logger.error("Trello is currently unavailable");
+                process.exit(1);
+            }
+
+            var foundCards = [];
+            for (var i in data) {
+                var card = data[i];
+
+                // We only want cards from the list we specified
+                if (card.idList != listId) {
+                    continue;
+                }
+
+                if (options.match == 'exact') {
+                    if (card.name == options.title || !options.title){
+                            foundCards.push(card);
+                    }
+                } else if (options.match == 'contains') {
+                    if (card.name.includes(options.title)) {
+                            foundCards.push(card);
+                    }
+                }
+            }
+
```

Comment:
There's lots of interactive parts to the next section. It would be good to detect if the user is running the script interactively before presenting them with options. If the output is being piped in to another process, the interactivity hurts the end user.

I'm all for using `List` and `prompt`, but only if the user has an option to perform the same action non-interactively.

This is how I'd handle it personally:

* Have it error if you use `exact` and there is > 1 result.
* Archive all matching cards if you use contains